src/bindings/python/tests/test_runtime/test_core.py

```diff
@@ -276,22 +255,25 @@ def test_query_model(device):
     assert [
         key for key in query_model.keys() if key not in ops_func_names
     ] == [], "Not all network layers present in query_model results"
-    assert next(iter(set(query_model.values()))) == device, "Wrong device for some layers"
+    assert device in next(iter(set(query_model.values()))), "Wrong device for some layers"
 
 
 @pytest.mark.dynamic_library()
 @pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU", reason="Device independent test")
-def test_register_plugin():
+def test_register_cpu_plugin():
     core = Core()
-    core.register_plugin("openvino_intel_cpu_plugin", "BLA")
+    if "Intel" in core.get_property("CPU", "FULL_DEVICE_NAME"):
+        core.register_plugin("openvino_intel_cpu_plugin", "BLA")
+    else:
+        core.register_plugin("openvino_arm_cpu_plugin", "BLA")
     model = core.read_model(model=test_net_xml, weights=test_net_bin)
-    exec_net = core.compile_model(model, "BLA")
-    assert isinstance(exec_net, CompiledModel), "Cannot load the network to the registered plugin with name 'BLA'"
+    compiled_model = core.compile_model(model, "BLA")
+    assert isinstance(compiled_model, CompiledModel), "Cannot load the network to the registered plugin with name 'BLA'"
 
 
 @pytest.mark.dynamic_library()
 @pytest.mark.skipif(os.environ.get("TEST_DEVICE", "CPU") != "CPU", reason="Device independent test")
-def test_register_plugins():
+def test_register_cpu_plugins():
     core = Core()
     if platform == "linux" or platform == "linux2":
```

Comment:
why `plugins_path` is not used here?
I suppose `plugins_path` needs to have not only arm / x86, but operating system as well.

Reply:
have rewritten `plugins_path` function. Now it generates .xml file